src/redux/books/books.js

```diff
@@ -14,19 +19,92 @@ export const removeBook = (payload) => ({
   payload,
 });
 
+// bookDisplay action creator
+export const bookDisplay = (payload) => ({
+  type: BOOKS_DISPLAYED,
+  payload,
+});
+
+export const loadingStart = (payload) => ({
+  type: LOADING_STARTED,
+  payload,
+});
+
+export const loadingEnd = (payload) => ({
+  type: LOADING_FINISHED,
+  payload,
+});
+
 // declare the initial state
-const initialState = [];
+const initialState = { books: [] };
+
+const initialLoading = { loading: false };
 
 // the reducer function for the books
-const booksReducer = (state = initialState, { type, payload }) => {
+export const booksReducer = (state = initialState, { type, payload }) => {
   switch (type) {
     case BOOK_ADDED:
-      return [...state, payload];
+      return { books: [...state.books, payload] };
     case BOOK_REMOVED:
-      return (state.filter(({ id }) => id !== payload.id));
+      return { books: (state.books.filter(({ id }) => id !== payload.id)) };
+    case BOOKS_DISPLAYED:
+      return { books: [...state.books, ...payload] };
     default:
       return state;
   }
 };
 
-export default booksReducer;
+// the reducer function for the loaders
+export const loadingReducer = (state = initialLoading, { type, payload }) => {
+  switch (type) {
+    case LOADING_STARTED:
+      return { ...state, loading: payload };
+    case LOADING_FINISHED:
+      return { ...state, loading: payload };
+    default:
+      return state;
+  }
+};
```

Comment:
### Optional suggestion. 

- [ ] I would recommend creating another folder to keep the reducers instead of implementing actions and reducers in the same file.

Reply:
Thanks for the great suggestion. I had it in mind but I was only following the instructions.